src/GraphQL/Server.hs

```diff
@@ -76,11 +73,34 @@ infixr 8 :<>
 data a :<|> b = a :<|> b
 infixr 8 :<|>
 
+
+-- Result collects errors and values at the same time unless a handler
+-- tells us to bail out in which case we stop the processing
+-- immediately.
+data Result a = Result [ResolverError] a deriving (Show, Functor)
+
+-- Aggregating results keeps all errors and creates a ValueList
+-- containing the individual values.
+aggregateResults :: [Result GValue.Value] -> Result GValue.Value
+aggregateResults r =
+  let (errs, valueList) = foldl' (\(eb, vb) (Result ea va) -> ((eb <> ea), va:vb)) ([], []) r
+  in Result errs (GValue.toValue valueList)
```

Comment:
This is just:

```haskell
GValue.toValue <$> sequenceA r
```

Which I figured out from the following logic:

```haskell
[Result GValue.Value] -> Result GValue.Value -- starting type
[Result a] -> Result a -- abstract 'Value'
[Result a] -> Result [a] -- replace second 'a' with list, because that's what it really is (Value masks things)
Applicative f => [f a] -> f [a] -- abstract 'Result' to 'Applicative'
(Applicative f, Traversable t) => t (f a) -> f (t a) -- abstract list to traversable
```

The logic might be wrong, but it's also what I'd expect to fall out of your very natural definition of applicative below.